lte/gateway/c/oai/tasks/mme_app/mme_app_state_converter.cpp

```diff
@@ -819,5 +820,34 @@ void MmeNasStateConverter::proto_to_ue(
     const oai::UeContext& ue_ctxt_proto, ue_mm_context_t* ue_ctxt) {
   proto_to_ue_mm_context(ue_ctxt_proto, ue_ctxt);
 }
+
+void MmeNasStateConverter::mme_app_ueip_imsi_map_to_proto(
+    const UeIpImsiMap* ueip_map, oai::MmeUeIpImsiMap* ueip_proto) {
+  auto proto_map = ueip_proto->mutable_mme_ueip_imsi_map();
+  for (auto itr = ueip_map->begin(); itr != ueip_map->end(); ++itr) {
```

Comment:
use `auto &` everywhere it makes sense unless you need a copy.

Reply:
same here, let's use for : each loop traversal here, and as @ulaskozat suggested, use `const auto&` as much as possible

Reply:
Thanks @ulaskozat and @ardzoht for suggestions and happy to learn new concepts